neutronless member tests need to clean up after themselves

* Title: [neutronless member tests](https://github.com/F5Networks/f5-openstack-agent/tree/master/test/functional/neutronless/member) need to clean up after themselves
* Attachments: 
* Details: 

##### Suggested Issue Type
Test-Error

##### Traceback
```python
$ py.test --symbols ../symbols.json.example ./member/ ./loadbalancer/
================================================ test session starts ================================================
platform linux2 -- Python 2.7.12, pytest-2.9.1, py-1.4.34, pluggy-0.3.1
rootdir: /home/ubuntu/git/f5-openstack-agent, inifile:
plugins: symbols-0.1.0a3, cov-2.2.1, f5-sdk-2.3.3
collected 19 items

member/test_single_member.py .E.E.E.E
member/test_single_member_opflex.py .E
member/test_single_member_weight_updates.py .
loadbalancer/test_f5_common_networks.py .s
loadbalancer/test_loadbalancer.py ..
loadbalancer/test_loadbalancer_admin_state.py .
loadbalancer/test_loadbalancer_disconnected.py ..
loadbalancer/test_loadbalancer_opflex.py ..
loadbalancer/test_loadbalancer_snat.py ...
loadbalancer/test_snat_common_network.py .

====================================================== ERRORS =======================================================
_________________________________ ERROR at teardown of test_create_single_member_up _________________________________

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def check_resulting_cfg(cls):
        """Check the current BIG-IP cfg agianst previous Reset upon Error

            This classmethod will check the current BIG-IP config and raise if
            there are any changes from the previous snap-shot.  Upon raise, the
            method will attempt to clear the BIG-IP back to the previous config
            """
>       cls._resulting_bigip_cfg()

bigip_interaction.py:126:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def _resulting_bigip_cfg(cls):
        """Raises upon polluted state and restores from backup"""
        result = cls._get_current_bigip_cfg()
        with open(cls.config_file) as fh:
            try:
                assert result == fh.read(), \
>                   "Test was unable to clean up BIG-IP cfg"
E                   AssertionError: Test was unable to clean up BIG-IP cfg

bigip_interaction.py:111: AssertionError
----------------------------------------------- Captured stderr setup -----------------------------------------------
ERROR:f5_openstack_agent.lbaasv2.drivers.bigip.esd_filehandler:Tag lbaas_policy failed validation for ESD esd_demo_1 and was not added to ESD. Error: Invalid value [u'demo_policy'] for tag lbaas_policy
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
--------------------------------------------- Captured stderr teardown ----------------------------------------------
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Saving active configuration...
Current configuration backed up to /var/local/ucs/cs_backup.ucs.
Product : BIG-IP
Platform: Z100
Version : 12.1.0
Edition : Hotfix HF1
Hostname: host-172.16.2.3.openstacklocal
Installing --full-- configuration on host host-172.16.2.3.openstacklocal
Installing configuration...
Installing the license file...
Post-processing...
usermod: no changes
Reloading License and configuration - this may take a few minutes...
________________________________ ERROR at teardown of test_create_single_member_down ________________________________

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def check_resulting_cfg(cls):
        """Check the current BIG-IP cfg agianst previous Reset upon Error

            This classmethod will check the current BIG-IP config and raise if
            there are any changes from the previous snap-shot.  Upon raise, the
            method will attempt to clear the BIG-IP back to the previous config
            """
>       cls._resulting_bigip_cfg()

bigip_interaction.py:126:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def _resulting_bigip_cfg(cls):
        """Raises upon polluted state and restores from backup"""
        result = cls._get_current_bigip_cfg()
        with open(cls.config_file) as fh:
            try:
                assert result == fh.read(), \
>                   "Test was unable to clean up BIG-IP cfg"
E                   AssertionError: Test was unable to clean up BIG-IP cfg

bigip_interaction.py:111: AssertionError
----------------------------------------------- Captured stderr setup -----------------------------------------------
ERROR:f5_openstack_agent.lbaasv2.drivers.bigip.esd_filehandler:Tag lbaas_policy failed validation for ESD esd_demo_1 and was not added to ESD. Error: Invalid value [u'demo_policy'] for tag lbaas_policy
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
--------------------------------------------- Captured stderr teardown ----------------------------------------------
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Saving active configuration...
Current configuration backed up to /var/local/ucs/cs_backup.ucs.
Product : BIG-IP
Platform: Z100
Version : 12.1.0
Edition : Hotfix HF1
Hostname: host-172.16.2.3.openstacklocal
Installing --full-- configuration on host host-172.16.2.3.openstacklocal
Installing configuration...
Installing the license file...
Post-processing...
usermod: no changes
Reloading License and configuration - this may take a few minutes...
______________________________ ERROR at teardown of test_create_single_member_down_up _______________________________

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def check_resulting_cfg(cls):
        """Check the current BIG-IP cfg agianst previous Reset upon Error

            This classmethod will check the current BIG-IP config and raise if
            there are any changes from the previous snap-shot.  Upon raise, the
            method will attempt to clear the BIG-IP back to the previous config
            """
>       cls._resulting_bigip_cfg()

bigip_interaction.py:126:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def _resulting_bigip_cfg(cls):
        """Raises upon polluted state and restores from backup"""
        result = cls._get_current_bigip_cfg()
        with open(cls.config_file) as fh:
            try:
                assert result == fh.read(), \
>                   "Test was unable to clean up BIG-IP cfg"
E                   AssertionError: Test was unable to clean up BIG-IP cfg

bigip_interaction.py:111: AssertionError
----------------------------------------------- Captured stderr setup -----------------------------------------------
ERROR:f5_openstack_agent.lbaasv2.drivers.bigip.esd_filehandler:Tag lbaas_policy failed validation for ESD esd_demo_1 and was not added to ESD. Error: Invalid value [u'demo_policy'] for tag lbaas_policy
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
--------------------------------------------- Captured stderr teardown ----------------------------------------------
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Saving active configuration...
Current configuration backed up to /var/local/ucs/cs_backup.ucs.
Product : BIG-IP
Platform: Z100
Version : 12.1.0
Edition : Hotfix HF1
Hostname: host-172.16.2.3.openstacklocal
Installing --full-- configuration on host host-172.16.2.3.openstacklocal
Installing configuration...
Installing the license file...
Post-processing...
usermod: no changes
Reloading License and configuration - this may take a few minutes...
______________________________ ERROR at teardown of test_create_single_member_up_down _______________________________

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def check_resulting_cfg(cls):
        """Check the current BIG-IP cfg agianst previous Reset upon Error

            This classmethod will check the current BIG-IP config and raise if
            there are any changes from the previous snap-shot.  Upon raise, the
            method will attempt to clear the BIG-IP back to the previous config
            """
>       cls._resulting_bigip_cfg()

bigip_interaction.py:126:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def _resulting_bigip_cfg(cls):
        """Raises upon polluted state and restores from backup"""
        result = cls._get_current_bigip_cfg()
        with open(cls.config_file) as fh:
            try:
                assert result == fh.read(), \
>                   "Test was unable to clean up BIG-IP cfg"
E                   AssertionError: Test was unable to clean up BIG-IP cfg

bigip_interaction.py:111: AssertionError
----------------------------------------------- Captured stderr setup -----------------------------------------------
ERROR:f5_openstack_agent.lbaasv2.drivers.bigip.esd_filehandler:Tag lbaas_policy failed validation for ESD esd_demo_1 and was not added to ESD. Error: Invalid value [u'demo_policy'] for tag lbaas_policy
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
--------------------------------------------- Captured stderr teardown ----------------------------------------------
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Saving active configuration...
Current configuration backed up to /var/local/ucs/cs_backup.ucs.
Product : BIG-IP
Platform: Z100
Version : 12.1.0
Edition : Hotfix HF1
Hostname: host-172.16.2.3.openstacklocal
Installing --full-- configuration on host host-172.16.2.3.openstacklocal
Installing configuration...
Installing the license file...
Post-processing...
usermod: no changes
Reloading License and configuration - this may take a few minutes...
_____________________________ ERROR at teardown of test_lbaas_service_one_opflex_member _____________________________

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def check_resulting_cfg(cls):
        """Check the current BIG-IP cfg agianst previous Reset upon Error

            This classmethod will check the current BIG-IP config and raise if
            there are any changes from the previous snap-shot.  Upon raise, the
            method will attempt to clear the BIG-IP back to the previous config
            """
>       cls._resulting_bigip_cfg()

bigip_interaction.py:126:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'neutronless.bigip_interaction.BigIpInteraction'>

    @classmethod
    def _resulting_bigip_cfg(cls):
        """Raises upon polluted state and restores from backup"""
        result = cls._get_current_bigip_cfg()
        with open(cls.config_file) as fh:
            try:
                assert result == fh.read(), \
>                   "Test was unable to clean up BIG-IP cfg"
E                   AssertionError: Test was unable to clean up BIG-IP cfg

bigip_interaction.py:111: AssertionError
----------------------------------------------- Captured stderr setup -----------------------------------------------
ERROR:f5_openstack_agent.lbaasv2.drivers.bigip.esd_filehandler:Tag lbaas_policy failed validation for ESD esd_demo_1 and was not added to ESD. Error: Invalid value [u'demo_policy'] for tag lbaas_policy
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
--------------------------------------------- Captured stderr teardown ----------------------------------------------
Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Warning: Permanently added '10.190.24.227' (ECDSA) to the list of known hosts.
Warning: Permanently added '172.16.1.131' (RSA) to the list of known hosts.
Saving active configuration...
Current configuration backed up to /var/local/ucs/cs_backup.ucs.
Product : BIG-IP
Platform: Z100
Version : 12.1.0
Edition : Hotfix HF1
Hostname: host-172.16.2.3.openstacklocal
Installing --full-- configuration on host host-172.16.2.3.openstacklocal
Installing configuration...
Installing the license file...
Post-processing...
usermod: no changes
Reloading License and configuration - this may take a few minutes...
======================== 18 passed, 1 skipped, 1 pytest-warnings, 5 error in 409.35 seconds =========================
```

##### Impacted Tests:
Only neutronless tests.  Nightly and Smoke are impacted here.  Smoke (not unit-based smoke/Travis tests) tests are currently disabled, though.

#### OpenStack Release
All Supported (liberty, mitaka, newton)

#### Agent Version
All

#### Operating System
nightly

--- a/test/functional/neutronless/member/test_single_member.py
+++ b/test/functional/neutronless/member/test_single_member.py
@@ -77,8 +77,8 @@ def check_member_down(member):
     assert member.ratio == 1
 
 
-def test_create_single_member_down_up(bigip,
-                              track_bigip_cfg,
+def test_create_single_member_down_up(track_bigip_cfg,
+                              bigip,
                               icd_config,
                               icontrol_driver):
 
@@ -109,8 +109,8 @@ def test_create_single_member_down_up(bigip,
     m = get_member_created(bigip, pool_name, member_name, folder)
     check_member_up(m)
 
-def test_create_single_member_up_down(bigip,
-                              track_bigip_cfg,
+def test_create_single_member_up_down(track_bigip_cfg,
+                              bigip,
                               icd_config,
                               icontrol_driver):
 

